### PR TITLE
Allow returning None fields

### PR DESCRIPTION
**Pull request by Gnouc**
Fixes https://github.com/pyeve/eve-sqlalchemy/issues/154

### CODE REVIEW

**Comment by dkellner** [comment]
As of #154, we know that Eve is actually sending `None` values. I'd prefer to do exactly the same and even remove the config item. Such a feature would have to be added in Eve directly - Eve-SQLAlchemy is "just" a data layer.

Can you rebase this on current master and add an integration test (see `eve_sqlalchemy.tests.integration`)? I will release 0.5.1 even just with this fix, as it breaks backwards compatibility in an unexpected way (before https://github.com/pyeve/eve-sqlalchemy/issues/154#issuecomment-338721959 I thought 0.5.0 is actually behaving like 0.4.1, my bad...).

**Comment by Gnouc** [comment]
@dkellner Make the changes.

The test will be failed for 3 tests in patch methods of eve itself. I think the problem is those tests, not eve-sqlalchemy.

I will raise the issue in eve repo.

File: eve_sqlalchemy/tests/integration/collection_class_set.py [hunk 1]
```diff
@@ -84,3 +85,9 @@ def test_patch_parent(self):
         self.assert200(status)
         response, _ = self.get('parents', item=2)
         self.assertEqual(response['children'], [3, 4])
+
+    def test_get_can_return_none_value(self):
+        response, status = self.get('parents/1')
+        self.assert200(status)
+        self.assertIn('none_field', response)
+        self.assertIsNone(response['none_field'])
```
Comment (dkellner) [inline eve_sqlalchemy/tests/integration/collection_class_set.py]: Consider moving this test to its own test case as it's not really related to collection classes. How about `tests/integration/get_none_values.py`?

File: eve_sqlalchemy/validation.py [hunk 2]
```diff
@@ -156,12 +156,11 @@ def _validate_dependencies(self, document, dependencies, field,
         dcopy = None
         if self._original_document:
             dcopy = copy.copy(document)
-            # Use dict_update and remove_none_values from utils, so existing
-            # values in document don't get overridden by the original document
-            # and None values are removed. Otherwise handling in parent method
+            # Use dict_update from utils, so existing
+            # values in document don't get overridden by the original document.
+            # Otherwise handling in parent method
```
Comment (dkellner) [inline eve_sqlalchemy/validation.py]: It's good you didn't forget about updating that comment, but please fix the formatting, too :).

**Comment by dkellner** [comment]
Ok, I'm fine with xfailing the two tests, but I fixed the third one. It actually discovered a real bug in the dependencies validation. Feel free to rebase / squash my commits.

File: eve_sqlalchemy/tests/patch.py [hunk 3]
```diff
@@ -27,6 +28,7 @@ def test_patch_defaults(self):
                                  {'title': None}, None)
         super(TestPatch, self).test_patch_defaults()
 
+    @pytest.mark.xfail(True, run=False, reason='not applicable to SQLAlchemy')
     def test_patch_defaults_with_post_override(self):
```
Comment (dkellner) [inline eve_sqlalchemy/tests/patch.py]: You can remove the implementation of the tests too, we will not need it anymore. See e.g. `test_patch_null_objectid`.

File: eve_sqlalchemy/tests/patch.py [hunk 4]
```diff
@@ -18,6 +18,7 @@ def test_patch_objectid(self):
     def test_patch_null_objectid(self):
         pass
 
+    @pytest.mark.xfail(True, run=False, reason='not applicable to SQLAlchemy')
     def test_patch_defaults(self):
```
Comment (dkellner) [inline eve_sqlalchemy/tests/patch.py]: You can remove the implementation of the tests too, we will not need it anymore. See e.g. `test_patch_null_objectid`.

**Comment by dkellner** [comment]
Merged after rebase and additional fixes.

**Pull request closed without merging**